esp-radio/src/wifi/mod.rs

```diff
@@ -3135,6 +3136,31 @@ impl WifiController<'_> {
         }
     }
 
+    /// Get the Access Point information of AP to which the device is associated with.
+    /// The value is obtained from the last beacon.
+    ///
+    /// <div class="warning">
+    ///
+    /// - Use this API only in STA or AP-STA mode.
+    /// - This API should be called after the station has connected to an access point.
+    /// </div>
+    ///
+    /// # Errors
+    /// This function returns [`WifiError::Unsupported`] if the STA side isn't
+    /// running. For example, when configured for AP only.
+    pub fn ap_info(&self) -> Result<AccessPointInfo, WifiError> {
+        if self.mode()?.is_sta() {
+            let mut record: MaybeUninit<include::wifi_ap_record_t> = MaybeUninit::uninit();
+            let record = unsafe { MaybeUninit::assume_init_mut(&mut record) };
```

Comment:
This is UB, you can't do this.

Reply:
No need, we'll do that, just don't introduce more code like it, please :)

Reply:
Please see https://github.com/esp-rs/esp-hal/pull/4408 - it's okay to pass a pointer like that to the C code, and assume C code has initialized the data if the call succeeds.